File: api/src/main/java/io/bunnies/baas/resources/types/GifMediaType.java
package io.bunnies.baas.resources.types;

import io.bunnies.baas.Constants;

public class GifMediaType extends ResourceMediaType {
    public static final String KEY = "GIF";

    public GifMediaType(String resourceID) {
        super(resourceID);
    }

    @Override
    public String constructUrl() {
        return Constants.MEDIA_BASE_URL + "gif/" + this.resourceID + ".gif";
    }
}

File: api/src/main/java/io/bunnies/baas/resources/types/GifMediaType.kt
package io.bunnies.baas.resources.types

class GifMediaType(resourceID: String, mediaBaseUrl: String) : ResourceMediaType(resourceID, mediaBaseUrl) {

    override fun constructUrl(): String {
        return this.mediaBaseUrl + "gif/" + this.resourceID + ".gif"
    }

    companion object {
        val KEY = "GIF"
    }
}
